### ISSUE TITLE
Tweak relayer strategy for confirmations

### ISSUE THREAD

**Issue by tomusdrw** [paritytech/parity-bridges-common#617]
Relayers strategy for confirmations are not incentive-aligned in case there are multiple relayers. Re-think relayers strategy of confirmations, to align their incentives (currently they confirm whatever they see on-chain, so they might not get any rewards for such delivery, however they might be unblocking the lane (i.e. delivering more messages in the future)).

**Comment by tomusdrw** [comment]
Related #983

**Comment by svyatonik** [comment]
`InstantCurrencyPayments` is gone (even though the case has been already covered there), this issue is irrelevant
